packages/realm/src/errors.ts

```diff
@@ -138,3 +141,52 @@ export class ClientResetError extends SyncError {
     };
   }
 }
+
+/**
+ * An error class that indicates that one or more object changes have been reverted by the server.
+ * This can happen when the client creates/updates objects that do not match any subscription, or performs writes on
+ * an object it didn't have permission to.
+ */
+export class CompensatingWriteError extends SyncError {
+  /**
+   * The array of compensating writes performed by the server.
+   */
+  public compensatingWrites: CompensatingWriteErrorInfo[] = [];
```

Comment:
This name seems a bit verbose.

```suggestion
  public infos: CompensatingWriteErrorInfo[] = [];
```

Reply:
Alternatively.

```suggestion
  public writes: CompensatingWriteErrorInfo[] = [];
```